assets/wire/components/ListViewControls.tsx

```diff
@@ -18,38 +21,75 @@ function ListViewControls({
     hideSearchAllVersions,
     searchAllVersions,
     toggleSearchAllVersions,
+    newItems,
+    fetchItems,
 }: any) {
-    return(
-        <div className="navbar navbar--flex navbar--small navbar--list-controls">
-            <div className="navbar__inner navbar__inner--end">
-                {hideSearchAllVersions ? null : (
-                    <SearchAllVersionsControl
-                        activeNavigation={activeNavigation}
-                        searchAllVersions={searchAllVersions}
-                        toggleSearchAllVersions={toggleSearchAllVersions}
-                    />
-                )}
-                {!hideNewsOnly && <NewsOnlyControl
-                    activeNavigation={activeNavigation}
-                    newsOnly={newsOnly}
-                    toggleNews={toggleNews}
-                />}
-                <span className="navbar__divider"></span>
-                <ListViewOptions setView={setView} activeView={activeView} />
-                {(!noNavigationSelected(activeNavigation) || (hideSearchAllVersions && hideNewsOnly)) ? null : (
-                    <div className="content-bar__right--mobile">
-                        <ListSearchOptions
-                            hideSearchAllVersions={hideSearchAllVersions}
-                            searchAllVersions={searchAllVersions}
-                            toggleSearchAllVersions={toggleSearchAllVersions}
-                            hideNewsOnly={hideNewsOnly}
-                            newsOnly={newsOnly}
-                            toggleNews={toggleNews}
-                        />
-                    </div>
-                )}
+    const Wrapper = ({children}: any) => {
+        if (children == null) {
+            return null;
+        }
+        return (
+            <div className="navbar navbar--flex navbar--small navbar--list-controls">
+                {children}
             </div>
-        </div>
+        );
+    };
+
+    const renderRefreshButton = () => {
+        return (newItems || []).length
+            ? (
+                <div className="navbar__inner navbar__inner--icon">
+                    <NewItemsIcon
+                        newItems={newItems}
+                        refresh={fetchItems}
+                    />
+                </div>
+            )
+            : null;
+    };
+
+    return (
+        <ResizeObserverComponent>
```

Comment:
use `WithScreenSizeObserver` - it reports changes to screen size while `ResizeObserverComponent` reports changes to a DOM element's size.